ledger/acctupdates.go

```diff
@@ -1208,6 +1212,90 @@ func (au *accountUpdates) lookupResource(rnd basics.Round, addr basics.Address,
 	}
 }
 
+func (au *accountUpdates) lookupAssetResources(rnd basics.Round, addr basics.Address, assetIDGT basics.AssetIndex, limit uint64, synchronized bool) (data []ledgercore.AssetResourceWithIDs, validThrough basics.Round, err error) {
```

Comment:
We decided to completely disregard deltas - the as of round returned is the latest from the database (there is a cached db round check) - I'm questioning the logic in that synchronized if/else condition in general.

Reply:
> there is a cached db round check

the check `_, err = au.roundOffset(rnd)` ensures rnd is within deltas range that is completely not needed since we want data for rounds <= au.cachedDBRound. Additionally deltas ignoring behavior not documented anywhere (plus there should be a test explicitly checking this).